dragonchain/lib/dto/bnb.py

```diff
@@ -0,0 +1,319 @@
+import requests
+
+from dragonchain import logger
+from dragonchain import exceptions
+from dragonchain.lib.dto import model
+
+
+DC_MAINNET_NODE = "http://10.2.1.197"  # Mainnet BNB
+DC_TESTNET_NODE = "https://data-seed-pre-0-s3.binance.org:443/"  # Testnet BNB
+# we currently don't have a private testnet node set up...
+
+MAINNET_RPC_PORT = ":27147/"
+MAINNET_API_PORT = ":1169/api/v1/"
+
+CONFIRMATIONS_CONSIDERED_FINAL = 1  # https://docs.binance.org/faq.html#what-is-the-design-principle-of-binance-chain
+BLOCK_THRESHOLD = 3  # The number of blocks that can pass by before trying to send another transaction
+SEND_FEE = 37500  # transfer fee fixed at 0.000375 BNB : https://docs.binance.org/trading-spec.html#current-fees-table-on-mainnet
+
+_log = logger.get_logger()
+
+
+def new_from_user_input(user_input: Dict[str, Any]) -> "BinanceNetwork":
+    """Create a new BinanceNetwork model from user input
+    Args:
+        user_input: User dictionary input (assumed already passing create_binance_interchain_schema)
+    Returns:
+        Instantiated BinanceNetwork client
+    Raises:
+        exceptions.BadRequest: With bad input
+    """
+    dto_version = user_input.get("version")
+    if dto_version == "1":
+        if not user_input.get("private_key"):
+            # We need to create a private key if not provided
+            wallet = Wallet.create_random_wallet()
+            user_input["private_key"] = wallet.private_key
+        else:
+            try:
+                # Check if user provided key is hex and convert if necessary
+                if len(user_input["private_key"]) == 66:  # private keys in hex are 66 chars with leading 0x
+                    user_input["private_key"] = user_input["private_key"][2:]  # Trim the 0x
+                if len(user_input["private_key"]) == 64:  # private keys in hex are 64 chars
+                    user_input["private_key"] = base64.b64encode(bytes.fromhex(user_input["private_key"])).decode("ascii")
+                else:  # assume key is a mnemonic string
+                    wallet = Wallet.create_wallet_from_mnemonic(user_input.get("private_key"))
+                    user_input["private_key"] = wallet.private_key
+            except Exception:
+                # If there's an error here, it's a bad key. Just set it to something bad as bad keys are caught later when making the client
+                user_input["private_key"] = "BAD_KEY_WAS_FOUND"
+        # check if it's hex...
+
+        # initialize from mem:
+        # wallet = Wallet.create_wallet_from_mnemonic('mnemonic word string', env=testnet_env)
+
+        # check for user-provided node address
+        if not user_input.get("node_ip"):
+            # default to Dragonchain managed Binance node if not provided
+            user_input["node_ip"] = DC_TESTNET_NODE if user_input["testnet"] else DC_MAINNET_NODE
+            user_input["rpc_port"] = MAINNET_RPC_PORT
+            user_input["api_port"] = MAINNET_API_PORT
+
```

Comment:
add an else here which checks that node_ip, rpc_port, and api_port all exist and are valid

Reply:
Added check to `ping()` to also check the API server.

Reply:
Fixed.